pytest_thawgun/plugin.py

```diff
@@ -47,53 +36,85 @@ def _datetime(self, current_time):
         while self.loop._ready:
             await asyncio.sleep(0)
 
+    def __enter__(self):
+        self._tick = False
+        self.set_wall_clock(to=datetime.now())
+        self._adjust_loop_clock(to=self.loop.time())
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._tick = True
+        self._adjust_loop_clock(to=self.loop.time())
+        self.set_wall_clock(to=datetime.now())
+
+    def _adjust_loop_clock(self, to):
+        if self._tick:
+            # Make it seem to the outside world as though the loop clock was resumed in an instant.
+            self._frozen_loop_time = None
+            loop_real_time_drift = self.loop._real_time() - to
+            self.loop.time = lambda: self.loop._real_time() - loop_real_time_drift
+        else:
+            self._frozen_loop_time = to
+            self.loop.time = lambda: self._frozen_loop_time
+
+    def set_wall_clock(self, to):
+        if self._freeze_time is not None:
+            self._freeze_time.stop()
+        self._freeze_time = freeze_time(to, tick=self._tick)
+        self._frozen_wall_clock_control = self._freeze_time.start()
```

Comment:
It seems we can return `self._freeze_time.start()` from this method - this will make class field `self._frozen_wall_clock_control` redundant.